### DISCUSSION TITLE
Existing user creates a new user

### DISCUSSION THREAD

**Discussion by fellowshipdevs** [supabase/supabase#3565] [Questions]
Is it possible to have an API endpoint where an authenticated user can create a new user? In my use case, I have a user type that should be able to create new users. These new users are then automatically assigned to the user who created them.

Currently, the signup endpoint is anonymous, so I am unable to pass in a token that would identify that the user who made the request is authenticated (and therefore be able to extract the uid() from the request.

**Comment by dshukertjr** [comment]
Hi @fellowshipdevs !

What kind of application are you trying to make by implementing this feature? I'm asking, because what you are describing might be solvable by storing the data in a separate users table or something.

**Reply by fellowshipdevs** [reply]
For example, in the app, I will have a user type (let's say "regional manager", or RM) that will manage their own list of another user type (let's say "local manager", or LM). I don't want the LMs to be able to sign up by themselves. I want the RMs to be able to add LMs to the app (with a generic password, for example), and then the LM can go into the app and change their password. 

However, in the backend, I want the LM to be tied only to the RM who created them. This way, when there are multiple RMs in the tool, they can only ever see the LM that they added themselves. 

Let me know if that makes a bit more sense.

**Reply by dshukertjr** [reply]
@fellowshipdevs thanks for the context. It helps a lot!

In that case, I might do it like this:

1. Create `regional manager` table and `local manager` table. In the `local manager` table, you have a column `regional manager id` as a foreign key to `regional manager` table. 
2. When a regional manager want to `sign up` a new local manager, they can send them an invitational link with [this method](https://github.com/supabase/gotrue-js/blob/master/src/GoTrueApi.ts#L219). 
3. Return value of the invitational link contains the newly created user. Use that information to create a new record inside local manager table. 

With this setup, you should be able to secure your app using [row level security](https://supabase.io/docs/guides/auth/row-level-security) as well to have different level of permissions depending on whether a user is a RM or LM. Does this look like what you were looking for?

**Reply by fellowshipdevs** [reply]
Thanks for the reply. I'm not sure how I would go about using that method. I'm building the front end with Draftbit, and using the APIs from Supabase (maybe I should have mentioned that initially, my bad). So I'm not really using packages.

In the Supabase API, I haven't seen an endpoint to invite by email. But even from your explanation, I still don't see how the LM is tied to the RM that creates/invites it. For example, If attaching the RM token to the signup endpoint worked, I would be able to get the auth.uid() from the API request by the RM, and create a database function that populates a table of RM.uid => LM.uid. But sadly, that doesn't work.

**Reply by dshukertjr** [reply]
[Here](https://supabase.io/docs/gotrue/server/about#post-invite) is the endpoint for sending invite. Note that you need to call this API from a backend with the service key. 

Once you invite a LM, you will get their user id as a return value. You can store that along with the user id of the inviter RM into the `local_managers` table as a single row. This way, each row in LM would have who their RM is. You can also get list of all of your LM if you are signed in as RM.

**Reply by jdcauley** [reply]
I think this is the thread I'm looking for but I was hoping to confirm.

I am working on a system where Stripe is where user account creation gets triggered.

The goal then being to send an email with a link for them to add a password/claim the account.

It sounds like triggering this endpoint to send that invite link would create that linkage and resolve my workflow.

Is that correct?
